src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1371 @@
+             s1*s0*s1*s2*s1*s2^-1*s1^4*s2*s1^-1*s2^-1*s1^-1*s0^-1*s1^-1,
+             s1*s0*s1*s2*s1*s2^-1*s1^-1*s2*s0^-1*s1^-1]
+            sage: A.braid_monodromy(vertical=True)
+            [s1*s0*s1*s0^-1*s1^-1*s0, s0^-1*s1*s0*s1^-1*s0, s0^-1*s1^2*s0]
+        """
+        if vertical:
+            computed = self._braid_monodromy_vertical
+        else:
+            computed = self._braid_monodromy_non_vertical
+        if computed is not None:
+            return computed
+        K = self.base_ring()
+        if not K.is_subring(QQbar):
+            raise TypeError('the base field is not in QQbar')
+        L = self.defining_polynomials()
+        bm, dic, dv, d1 = braid_monodromy(prod(L), arrangement=L,
+                                          vertical=vertical)
+        if vertical:
+            self._braid_monodromy_vertical = bm
+            self._strands_vertical = dic
+            self._vertical_lines_in_braid_mon = dv
+        else:
+            self._braid_monodromy_non_vertical = bm
+            self._strands_nonvertical = dic
+        return bm
+
+    def strands(self):
+        r"""
+        Return the strands for each member of the arrangement.
+
+        OUTPUT:
+
+        A dictionary which associates to the index of each strand
+        its associated component if the braid monodromy has been
+        calculated with ``vertical=False``.
+
+        .. NOTE::
+
+           This functionality requires the ``sirocco`` package to be installed.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 + x, y + x - 1, x)
+            sage: bm = A.braid_monodromy()
+            sage: A.strands()
+            {0: 2, 1: 1, 2: 0, 3: 0}
+        """
+        if not self._strands_nonvertical:
+            self._braid_monodromy = self.braid_monodromy(vertical=False)
+        return self._strands_nonvertical
+
+    def vertical_strands(self):
+        r"""
+        Return the strands if the braid monodromy has been computed with
+        the vertical option.
+
+        OUTPUT:
+
+        A dictionary which associates to the index of each strand
+        its associated component if the braid monodromy has been
+        calculated with ``vertical=True``.
+
+        .. NOTE::
+
+           This functionality requires the ``sirocco`` package to be installed.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 + x, y + x - 1, x)
+            sage: A.vertical_strands()
+            {0: 1, 1: 0, 2: 0}
+            sage: A.braid_monodromy(vertical=True)
+            [s1*s0*s1*s0^-1*s1^-1*s0, s0^-1*s1*s0*s1^-1*s0, s0^-1*s1^2*s0]
+        """
+        if not self._strands_vertical:
+            self.braid_monodromy(vertical=True)
+        return self._strands_vertical
+
+    def vertical_lines_in_braid_mon(self):
```

Comment:
```suggestion
    def vertical_lines_in_braid_monodromy(self):
```
It is better to spell everything out explicitly in the public methods.